scripts: add a simple script to quickly upload releasenotes

KanjiMonster opened the pull request:
Add a simple script to quickly upload an existing release-notes.txt for
all available platforms to repo.

Signed-off-by: Jonas Gorski <jonas.gorski@bisdn.de>

jklare commented on scripts/upload-releasenotes.sh:
```diff
@@ -0,0 +1,29 @@
+#!/bin/bash
+BRANCH="$(git symbolic-ref --short HEAD)"
+NOTES=${1:-release-notes.txt}
```
```suggestion
NOTES=${1:-releasenotes.txt}
```

KanjiMonster replied:
`release-notes.txt` is what we have used most of the time. I can change it to releasenotes.txt though.

jklare commented on scripts/upload-releasenotes.sh:
```diff
@@ -0,0 +1,29 @@
+PLATFORMS="$(grep '#MACHINE ?=' conf/local.conf.sample | awk '{ print $3 }' | tr -d '"')"
+
+for platform in $PLATFORMS; do
+	aws s3 cp $NOTES s3://repo.bisdn.de/pub/onie/$platform/onie-bisdn-$platform-$BRANCH-releasenotes.txt
```
```suggestion
	aws s3 cp $NOTES s3://repo.bisdn.de/pub/onie/$platform/onie-bisdn-$platform-$BRANCH-$NOTES
```

KanjiMonster replied:
This one should stay a hardcoded releasenotes.txt, as this is what we have always used. The source filename should not have any influence on the target filename.

KanjiMonster commented:
Updated with a newer version:

- no magic hardcoded filename, now needs a filename as an argument
- clarified the error messages a bit
- dropped the special handling of master (because its also not a release branch)
- dropped the grep usage by making awk match

jklare commented on scripts/upload-releasenotes.sh:
```diff
@@ -0,0 +1,29 @@
+#!/bin/bash
+BRANCH="$(git symbolic-ref --short HEAD)"
+TOPDIR=$(git rev-parse --show-toplevel)
+NOTES=$1
```
```suggestion
NOTES="releasenotes.txt"
```

jklare replied:
We should fix the name here and also align this with the docs. I pushed an update to the wiki on how to name the releasenotes.txt, maybe you can push one on how to use this script?

KanjiMonster replied:
NAK with this change. Let's not use magic file names at all, and require passing an argument.

jklare replied:
I think we should not use variables for things that are not meant to be variable.

jklare replied:
given that all involved files (releasenotes.txt and this script) will be accessible publicly, it should also be pretty straightforward to run this script directly as part of the release pipeline

KanjiMonster replied:
That would make the release process even longer slower than it currently is, as it means that:

a) the release notes file must be merged before we can start the release builds
b) the `default.xml` needs to be updated to point to the commit that adds the release notes, else it won't exist for the builder, which requires yet another PR

So I don't think this is a good idea the way we currently do the releases, as it will add more places where we can accidentally break the build. And we only would see this at the very end, wasting a full day just to find out we forgot something there.

The release process currently already is much too slow, and we should work on making it faster, not requiring more steps.

So I really don't think hardcoding a filename here is a good idea, as it just makes the release process more annoying, and the script is intended to make it easier, not harder.

jklare replied:
I think it would still be fine to start the builds before the releasenotes are merged, we just have to make sure that they are there before the pipeline gets to the final publish step. Adding the automatic upload would probably also simplify the process and make it a bit faster.

KanjiMonster replied:
The build process does only one checkout, at the beginning of the build process. Any pushes after that will have no effect on the pipeline, and thus pushing the release notes later will have no effect on it. There is no other way to wait for the release notes and the default.xml update to be merged, unless we completely rework how we do the checkout and build.

It would also break the assumption that the default.xml in the repository is what was used for building.

jklare replied:
Currently there is not direct relation between the initial checkout used for building the image and the final copy step in which the images are published during a release. The only connection there, is the $BRANCH we use to upload the images before testing. We can easily use the same branch and just copy one more file, we do not even need any additional git checkout for that (not that there is one during our copy stage anyways).

KanjiMonster replied:
The file won't exist for the image builder if pushed after the build was started. There will be no source file to copy from.

KanjiMonster replied:
Also maybe not try to change the release / build process during a release?

KanjiMonster replied:
Anyway, I don't think this will lead anywhere.

jklare replied:
I agree, we should not change anything during the release, but that was never my intention. I think this MR needs more work and is possibly not needed at all, since this can be done automatically during the copy stage of our pipelines.
